### DISCUSSION TITLE
I have all google speakers. Should I disable the HA Media Player Provider?

### DISCUSSION THREAD

**Discussion by ncd7** [music-assistant/support#2281] [Q&A] [answered]
After installing MA, I now have 3 versions of each media player.
One is from the HA Chromecast integration.
One is from the MA HA Player.
One is from the MA Chromecast integration.

Is it reasonable to disable the MA HA Player and only keep the MA Chromecast integration and the native HA Chromecast integration? Or possibly even only keep the MA Chromecast? 

Trying to figure out the pros and cons and keeping the least number of duplicates, or ideally no duplicates at all. Or is the best pattern to keep multiple copies with slightly different naming conventions? 

Thanks!

**Comment by OzGav** [comment]
I don’t see a reason that you need any of the players except the MA ones (now that we are at stable)

**Reply by ncd7** [reply]
Thanks for the response. You mean the MA HA Player?
I thought I read in the documentation that the MA Chromecast have better features and can play at a higher quality since the MA HA Player is sort of the least common denominator. 
Is that inaccurate? Thanks!

**Reply by OzGav** [reply]
No disable all EXCEPT the MA ones.

**Reply by ncd7** [reply]
Apologies but your statement is still a bit ambiguous. Which MA ones, there are multiple.

Keep the MA HA ones? Or keep the MA Chromecast ones? Or keep all MA ones? 
The MA documentation seems to indicate the MA HA ones might not support all features but perhaps that's outdated.

**Reply by OzGav** [reply]
OK go to HA SETTINGS>>DEVICES&SERVICES>>INTEGRATIONS>>MUSIC ASSISTANT and then click on the "xx entities" and you will see all the MA entities. If there are duplicates in there then you need to tidy that up in MA as you must have a player which supports multiple protocols. Any HA entity which represents the same device as one of the those MA entities you should be safe to disable (or delete if that what you want to do)

**Reply by ncd7** [reply]
EDIT: I meant to say provider, not plugin.

Yes, I think the easiest thing is to actually go in the configuration of MA directly.
I've learned that I can just disable a specific player provider and that will disable all the associated entities.

My question is -- should I keep both the MA Chromecast player provider and the MA HA player provider ? All my physical speakers in the home are google assistant based ones so they all support Chromecast. So for this case, do you think it's ok to just disable the MA HA provider , or is it better to keep it for some reason? 

![image](https://github.com/music-assistant/hass-music-assistant/assets/2722344/d8664274-af36-403f-9a1b-e7b311bed266)

**Reply by OzGav** [reply, marked as answer]
No you should only use the HA plugin if there is no native MA player available. So just use the Chromecast player provider as you suggested. You can remove Airplay, Sonos and DLNA as well if you aren't using them

**Reply by ncd7** [reply]
Got it, thanks so much!!
So if I remove the HA plugin players (and the other ones I don't need) and I also disable the Chromecast integration that comes with HA itself, the only set of players I'll have are the MA Chromecast ones. This is great! Thanks for the clarifications.
